packages/react/src/textarea/Textarea.tsx

```diff
@@ -0,0 +1,77 @@
+import { type ComponentPropsWithRef, type ReactNode, forwardRef } from "react";
+import TextareaAutosize from "react-textarea-autosize";
+
+import type { ExtendProps } from "../utils";
+
+import { Box } from "../box";
+import { Flex } from "../flex";
+import { extractSprinkles } from "../sprinkles";
+import * as styles from "./Textarea.css";
+
+type TextareaProps = ExtendProps<
+  ComponentPropsWithRef<"textarea">,
+  ComponentPropsWithRef<typeof Box>,
+  {
+    autoResize?: boolean;
+    bottomSection?: ReactNode;
+    disabled?: boolean;
+    error?: boolean;
+    readonly?: boolean;
+    resize?: "none" | "vertical";
+    topSection?: ReactNode;
+  }
+>;
+
+export const Textarea = forwardRef<HTMLTextAreaElement, TextareaProps>(
+  (
+    {
+      autoResize = false,
+      bottomSection,
+      className,
+      disabled,
+      error,
+      placeholder,
+      readOnly,
+      resize = "vertical",
+      rows,
+      topSection,
+      ...props
+    },
+    ref,
+  ) => {
+    const Component = autoResize ? TextareaAutosize : "textarea";
+    const { restProps, sprinkleProps } = extractSprinkles(props);
+
+    return (
+      <Flex
+        aria-disabled={disabled}
+        aria-invalid={error}
+        data-disabled={disabled}
+        data-invalid={error}
+        style={{
+          resize: resize === "none" ? "none" : "vertical",
+        }}
```

Comment:
Can we add style with 
```
export const resizeNone = recipe({
  base: [
    style({
      resize: "none",
    }),
  ],
});

export const resizeVertical = recipe({
  base: [
    style({
      resize: "vertical",
    }),
  ],
});

```

and use clsx to merge the style instead of using style in tsx?

Reply:
I guess, this is simpler in this manner.

Reply:
we should never use inline `styles` unless there's a very specific reason - this should be moved to `.css.ts`

Reply:
understood bhai. Addressed this using `data-size` attribute